File: tests/fixtures/golang/simple-functions.go
package main

import "fmt"

// Simple function with return type
func greet(name string) string {
	fmt.Println("Hello, " + name)
	return "Hello, " + name
}

// Function with multiple return values
func divide(a, b float64) (float64, error) {
	if b == 0 {
		return 0, fmt.Errorf("division by zero")
	}
	return a / b, nil
}

// Variadic function
func sum(numbers ...int) int {
	total := 0
	for _, n := range numbers {
		total += n
	}
	return total
}

// Function with named return values
func createUser(name string, age int) (user string, err error) {
	if name == "" {
		err = fmt.Errorf("name required")
		return
	}
	user = fmt.Sprintf("%s (age %d)", name, age)
	return
}

// Exported function
func FormatDate(year, month, day int) string {
	return fmt.Sprintf("%04d-%02d-%02d", year, month, day)
}

File: tests/fixtures/rust/simple-functions.rs
// Simple function with return type
fn greet(name: &str) -> String {
    println!("Hello, {}", name);
    format!("Hello, {}!", name)
}

// Function with multiple parameters
fn add(a: i32, b: i32) -> i32 {
    a + b
}

// Function with Result return type
fn divide(a: f64, b: f64) -> Result<f64, String> {
    if b == 0.0 {
        Err("division by zero".to_string())
    } else {
        Ok(a / b)
    }
}

// Generic function
fn largest<T: PartialOrd>(list: &[T]) -> &T {
    let mut largest = &list[0];
    for item in &list[1..] {
        if item > largest {
            largest = item;
        }
    }
    largest
}

// Async function
async fn fetch_data(url: &str) -> Result<String, Box<dyn std::error::Error>> {
    println!("Fetching: {}", url);
    Ok(String::from("response"))
}

// Public function
pub fn format_date(year: u32, month: u32, day: u32) -> String {
    format!("{:04}-{:02}-{:02}", year, month, day)
}
